source/fundamentals/context.txt

```diff
@@ -1 +1,119 @@
-TODO: Discussion of the ``context`` package
\ No newline at end of file
+=======
+Context
+=======
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how the MongoDB Go Driver uses the
+**context package** from Go's standard library. 
+
+The context package consists of the ``Context`` and ``CancelFunc``
+types. The Go Driver uses the ``Context`` type to simplify the
+process of creating timeout or cancellation tokens for an operation in
+a blocking function. A blocking function is a function that relies
+on an event, such as an input or output, to proceed with its task.
+
+Expiration
+----------
+
+The driver considers a context expired when an operation exceeds its
+timeout or cancellation. It checks the context expiration with the ``Done()``
+function of the ``Context`` type. The ``Done()`` function returns a
+closed channel and when the function should cancel the work completed.
+If you can successfully read from the channel, then the context expired.
+
+The following sections show when and how the driver checks for
```

Comment:
Suggestion: We don't have any diagrams or screenshots here, so I think "describe" is more accurate than "show".

```suggestion
The following sections describe when and how the driver checks for
```